preview/MsixCore/msixmgr/BcryptLibrary.cpp

```diff
@@ -0,0 +1,61 @@
+// Copyright (C) Microsoft. All rights reserved.
+
+#include <Windows.h>
+#include <WinCrypt.h>
+#include <BcryptLibrary.hpp>
+#include <assert.h>
+
+/// @note We intentionally load BCRYPT once on first use and never explicitly unload it;
+///       it's intentionally kept loaded until process termination unloads it for us.
+///
+/// @note We initialize the module handle to nullptr @ compile-time and only Load() modifies it,
+///       and then only on first need.
+///
+/// @note In the exceedingly rare event 2 threads concurrently call Load(), we'd call
+///       LoadLibrary() twice but producing the same result (the 2nd call finds the DLL already
+///       in the LoadedModuleTable and returns the same handle after bumping its reference
+///       count). Thus we don't need any special synchronization to manage this, as worst case
+///       we'll wind up overwriting the global with the same value on the 2nd load.
+volatile PVOID MsixCoreLib::BcryptLibrary::bcryptModule = nullptr;
+FARPROC MsixCoreLib::BcryptLibrary::functions[MsixCoreLib::BcryptLibrary::FunctionCount];
+PCSTR MsixCoreLib::BcryptLibrary::functionNames[MsixCoreLib::BcryptLibrary::FunctionCount] = {
+    "BCryptOpenAlgorithmProvider",
+    "BCryptCloseAlgorithmProvider",
+    "BCryptGetProperty",
+    "BCryptCreateHash",
+    "BCryptHashData",
+    "BCryptFinishHash",
+    "BCryptDestroyHash"
+};
+
+///TODO:5204957 MsixCoreLib::BcryptLibrary::Load() (initialization) leaks

```

Comment:
>TODO [](start = 3, length = 4)

? comment's not particularly useful. sounds like we're leaking this on purpose..